tests/flow/test_with_clause.py

```diff
@@ -241,3 +242,13 @@ def test10_filter_placement_validate_scopes(self):
         actual_result = redis_graph.query(query)
         expected = [['projected']] # The projected string should be returned
         self.env.assertTrue(re.search('Filter\s+Project', plan))
+
+    def test11_valid_order_by_aliases(self):
+        # Verify that ORDER BY aliases match previously defined references
+        query = """UNWIND [1,2,3] AS a WITH a ORDER BY nonexistent RETURN a"""
```

Comment:
What about the valid case? `"UNWIND [1,2,3] AS a WITH a ORDER BY a RETURN a"`
and a mixture, `"UNWIND [1,2,3] AS a WITH a ORDER BY a, nonexistent RETURN a"`
Lastly order by a variable which is out of scope

```
UNWIND [1,2,3] AS a WITH a
UNWIND [1,2,3] AS b WITH b ORDER BY a
RETURN b
```

Reply:
Your last example is valid since `a` is still in scope, but I added the requested tests with a slight variation on it.